src/OSPSuite.Core/Commands/UpdateValueOriginCommand.cs

```diff
@@ -0,0 +1,55 @@
+﻿using OSPSuite.Core.Commands.Core;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using Command = OSPSuite.Assets.Command;
+
+namespace OSPSuite.Core.Commands
+{
+   public class UpdateValueOriginCommand : OSPSuiteReversibleCommand<IOSPSuiteExecutionContext>
```

Comment:
I have created a command so that undoing the action of setting the PI parameter will also undo the automatic set.
I would not create a command however for VAlueOrigin set manually (e.g. by the user as opposed to programatically)

Reply:
>I would not create a command however for VAlueOrigin set manually (e.g. by the user as opposed to programatically)

Why not? If a value is reseted to an older one, Value Origin should be reseted too, don't you think?

Reply:
I would create a command ONLY for value origin set automatically. Right now we do not have a command for Value Description and I think this is correct (Commands are only for actions that change the model output)

In this case, we update the valueorigin in code (without the user knowing about it). Undo (and not reset) should also undo the automatic change

Reply:
ok, let's accept it for the moment.
might need further discussion in the future